### PR TITLE
Add a configuration view

### PR DESCRIPTION
**Pull request by orlandohohmeier**
---

ℹ️   _This branch depends on changes that will be introduced with #236, thus it will fail until the respective branch is merged._

---
Introduce a config/version view. This component will be used in the service detail configuration tab. These commits are cherry picked from #155 in order to reduce the PR size.

_Integrated config view_

### CODE REVIEW

File: src/js/components/ConfigurationView.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    fetchVersion(service, versionID);
+  }
+
+  componentWillReceiveProps({service, versionID}) {
+    super.componentWillReceiveProps(...arguments);
+
+    fetchVersion(service, versionID);
+  }
+
+  getGenralSection({id, cmd, cpus, disk, instances, mem}) {
```
Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: General?

File: src/js/components/ConfigurationView.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+class ConfigurationView extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.store_listeners = [{name: 'dcos', events: ['change']}];
+  }
+
+  componentWillMount() {
+    super.componentWillMount();
```
Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: So to understand: when using `reactjs-mixin` we need to always remember to call `super.[lifeCycleMethod]`?

Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: Actually, I don't see why this one is needed. In any case, StoreMixin only defines `didMount` and `willUnmount` LF methods.

File: src/js/components/ConfigurationView.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+    let {service, versionID} = this.props;
+    fetchVersion(service, versionID);
+  }
+
+  componentWillReceiveProps({service, versionID}) {
+    super.componentWillReceiveProps(...arguments);
```
Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: Same here. I feel lost 👻

File: src/js/components/ConfigurationView.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+function fetchVersion(service, versionID) {
+  if (service.getVersions().get(versionID) == null) {
+    DCOSStore.fetchServiceVersion(service.getId(), versionID);
+  }
+}
+
+class ConfigurationView extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
```
Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: I'd like to know the case for passing `...arguments` here VS just `super()`.

File: src/js/components/ConfigurationView.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+          case 'COMMAND':
+            headerValueMapping = Object.assign(headerValueMapping, {
+              'Command': healthCheck.command.value,
+            });
+            break;
+          case 'HTTP':
+            headerValueMapping = Object.assign(headerValueMapping, {
+              'Path': healthCheck.path,
+              'Port': healthCheck.port,
+              'Port Index': healthCheck.portIndex || 'PORT_INDEX',
```
Comment (pierlo-upitup) [inline src/js/components/ConfigurationView.js]: hat does `PORT_INDEX` mean?

File: src/js/utils/StringUtil.js [hunk 6]
```diff
@@ -1,6 +1,14 @@
 import marked from 'marked';
 
 const StringUtil = {
+  arrayToCommaSeparatedString: function (array) {
```
Comment (pierlo-upitup) [inline src/js/utils/StringUtil.js]: I don't see a huge need for this function, to be honest – but I guess we could make it a tad more generic for example by having the `separator` be an optional argument?
How about `arrayToJoinedString(array = [], separator = ',') { ` ?

**Comment by pierlo-upitup** [comment]
Cosmetics/nits – rest LGTM

File: src/js/components/ConfigurationView.js (hunk 4 above)
Comment (orlandohohmeier) [inline src/js/components/ConfigurationView.js]: The `...arguments` kinda makes sure, that this component is future proof and we're not missing something if we update one of the parent classes. 
I don't think that we have a rule for these yet, thus I suggest to discuss this in our next sync meeting. 

/cc @rcorral /link https://github.com/dcos/dcos-ui/pull/207#discussion_r63930608

File: src/js/components/ConfigurationView.js (hunk 5 above)
Comment (orlandohohmeier) [inline src/js/components/ConfigurationView.js]: Upsi, that's is actually a bug. I somehow screwed that one. I blame the jetlag... The `PORT_INDEX` is actually the default `healthCheck.portType` value which defines the `portIndex` value type.

File: src/js/utils/StringUtil.js (hunk 6 above)
Comment (orlandohohmeier) [inline src/js/utils/StringUtil.js]: Good call. I only extracted that one in to a function as I was repeating myself over and over again. I'll generalize the method.

**Comment by pierlo-upitup** [comment]
#220 has landed, please rebase etc.

File: src/js/components/ConfigurationView.js (hunk 2 above)
Comment (rcorral) [inline src/js/components/ConfigurationView.js]: This is a pattern we've been doing for a very long time now. It makes it easy to add new mixins without having to think about which methods each of them implement.
The overhead for the reward is negligible.

**Pull request merged**